app/controllers/frontend_log_controller.rb

```diff
@@ -77,4 +77,15 @@ def valid_event?
   def valid_payload?
     !log_params[:payload].nil?
   end
+
+  def hash_from_method_kwargs(hash, method)
+    method_kwargs(method).index_with { |key| hash[key.to_s] }
+  end
+
+  def method_kwargs(method)
+    method.
+      parameters.
+      select { |info_pair| [:key, :keyreq].include?(info_pair[0]) }.
```

Comment:
The possible values for the parameter "type" aren't really well documented.

https://ruby-doc.org/core-2.5.1/Method.html#method-i-parameters

From [this blog post](https://jemma.dev/blog/ruby-method-parameters), I think this covers what we care about though.

```
[:req, :required_positional]
[:opt, :optional_positional]
[:rest, :splat_args]
[:keyreq, :required_keyword]
[:key, :optional_keyword]
[:keyrest, :double_splat_args]
[:block, :block]
```